CliWrap/Command.Execution.cs

```diff
@@ -171,30 +171,215 @@ await StandardInputPipe
         }
     }
 
-    private async Task PipeStandardOutputAsync(
-        ProcessEx process,
+    private async Task PipeStreamAsync(
+        Stream source,
+        PipeTarget target,
         CancellationToken cancellationToken = default
     )
     {
-        await using (process.StandardOutput.ToAsyncDisposable())
+        await using (source.ToAsyncDisposable())
         {
-            await StandardOutputPipe
-                .CopyFromAsync(process.StandardOutput, cancellationToken)
-                .ConfigureAwait(false);
+            await target.CopyFromAsync(source, cancellationToken).ConfigureAwait(false);
         }
     }
 
-    private async Task PipeStandardErrorAsync(
-        ProcessEx process,
+    // PTY-specific piping methods
+
+    private async Task PipePtyInputAsync(
+        PtyProcessEx process,
         CancellationToken cancellationToken = default
     )
     {
-        await using (process.StandardError.ToAsyncDisposable())
+        // IMPORTANT: Do NOT dispose the stdin stream here!
+        // With PTY, closing the input pipe causes ConPTY to send Ctrl+C to the process.
+        // The stream will be cleaned up when the PTY is disposed after the process exits.
+        try
         {
-            await StandardErrorPipe
-                .CopyFromAsync(process.StandardError, cancellationToken)
+            await StandardInputPipe
+                .CopyToAsync(process.StandardInput, cancellationToken)
                 .ConfigureAwait(false);
+
+            // Flush to ensure data is sent to the PTY immediately
+            await process.StandardInput.FlushAsync(cancellationToken).ConfigureAwait(false);
+        }
+        catch (IOException ex) when (ex.GetType() == typeof(IOException)) { }
```

Comment:
Can you please add comments describing when each ignored exception type can happen and why we're ignoring them. Same for the `PipePtyOutputAsync` method.